File: backend/src/Endurist.Core/Activities/Models/ActivityModel.cs
using SideEffect.DataTransfer.Models;

namespace Endurist.Core.Activities.Models;

public class ActivityModel : ActivityPreviewModel
{
    public List<SegmentPreviewModel> Segments { get; set; }

    public List<KeyValueModel<double, int?>> HeartRateGraph { get; set; }

    public List<KeyValueModel<double, int>> PaceGraph { get; set; }

    public List<KeyValueModel<double, double?>> AltitudeGraph { get; set; }

    public List<KeyValueModel<double, int?>> CadenceGraph { get; set; }
}

File: frontend/src/interfaces/activities/ActivityModel.ts
import ActivityPreviewModel from "./ActivityPreviewModel";
import KeyValueModel from "../KeyValueModel";
import SegmentPreviewModel from "../SegmentPreviewModel";

interface ActivityModel extends ActivityPreviewModel {
  segments: SegmentPreviewModel[];
  heartRateGraph: KeyValueModel<number, number | null>[];
  paceGraph: KeyValueModel<number, number>[];
  altitudeGraph: KeyValueModel<number, number | null>[];
  cadenceGraph: KeyValueModel<number, number | null>[];
}

export default ActivityModel;
